drivers/can/can_nxp_s32_canxl.c

```diff
@@ -829,6 +829,11 @@ static int can_nxp_s32_init(const struct device *dev)
 		}
 	}
 
+	if (!device_is_ready(config->clock_dev)) {
+		LOG_ERR("Clock control device not ready");
+		return -ENODEV;
+	}
```

Comment:
Shouldn't a call to `clock_control_on()` be made here?

Reply:
you're right, although the default configuration have the needed clock on at init. Let's add it for completeness